integration/v3_grpc_test.go

```diff
@@ -491,3 +491,86 @@ func TestV3WatchCancel(t *testing.T) {
 
 	clus.Terminate(t)
 }
+
+// TestV3WatchMultiple tests multiple watchers on the same key
+// and one watcher with matching prefix. It first puts the key
+// that matches all watchers, and another key that matches only
+// one watcher to test if it receives expected events.
+func TestV3WatchMultiple(t *testing.T) {
+	clus := newClusterGRPC(t, &clusterConfig{size: 3})
+	wAPI := pb.NewWatchClient(clus.RandConn())
+	kvc := pb.NewKVClient(clus.RandConn())
+
+	wStream, errW := wAPI.Watch(context.TODO())
+	if errW != nil {
+		t.Fatalf("wAPI.Watch error: %v", errW)
+	}
+
+	go func() {
+		if _, err := kvc.Put(context.TODO(), &pb.PutRequest{Key: []byte("foo"), Value: []byte("bar")}); err != nil {
+			t.Fatalf("couldn't put key (%v)", err)
+		}
+	}()
+
+	keysToWatch := []string{"fo", "foo", "foo", "foo", "foo"}
+	keysToReceive := []string{"foo", "foo", "foo", "foo", "foo"}
+
+	for i := range keysToWatch {
+		go func(key string) {
+			if err := wStream.Send(&pb.WatchRequest{CreateRequest: &pb.WatchCreateRequest{Prefix: []byte(key)}}); err != nil {
```

Comment:
This line contradicts with what describes in the comment. All requests create watcher on prefix not keys. The description says only one should watch on prefix, and I assume it is `fo` one. Let's break this up to a simple watch on prefix and a for loop for watching on key.